Slightly optimize computation for `RigidContacts` model

flferretti opened the pull request:
This PR slightly enhances the performances of the `RigidContacts` model by using the pseudoinverse instead of the least-squares method to compute the impact velocity, and removes some unnecessary context managers and assertions

CarlottaSartore commented:
We checked with @flferretti  and the time needed to perform `jnp.linalg.pinv` and `jnp.linalg.lstsq`  are comparable but since `lstq` is more stable for ill conditioned matrix , it could have sense to keep the original implementation. 

Maybe we can close this PR then @flferretti  ?

xela-95 reviewed (changes requested):
Thanks @flferretti ! I left a comment

xela-95 commented:
> We checked with @flferretti and the time needed to perform `jnp.linalg.pinv` and `jnp.linalg.lstsq` are comparable but since `lstq` is more stable for ill conditioned matrix , it could have sense to keep the original implementation.
> 
> Maybe we can close this PR then @flferretti ?

Sorry @CarlottaSartore I skipped your comment! Mmmh yes, then maybe we can close this for now. Do you agree @flferretti ?

flferretti commented:
> Sorry @CarlottaSartore I skipped your comment! Mmmh yes, then maybe we can close this for now. Do you agree @flferretti ?



> Maybe we can close this PR then @flferretti ?

Sure! I also removed a switch of the velocity representation and refactored a bit the function. If you agree I can restore the `jnp.linalg.solve` and keep the rest as it is
